SPEC.md

```diff
@@ -0,0 +1,282 @@
+# Backend API Spec
+
+## Content Type
+
+### Requests
+
+The API uses JSON for requests and responses. This means that all requests
+that contain a payload (`POST` and `PUT` requests) must have the `Content-Type` and `Accept` headers set to `application/json`.
+
+### Responses
+
+All successful API calls return as content type `application/json`,
+except the `GET /image/<digest>` request.
+
+```
+HTTP/1.0 200 OK
+Content-Length: ...
+Content-Type: application/json
+...
+```
+
+## Posts
+
+### `POST /posts`
+
+Create a new post.
+
+#### Request:
+
+```json
+{
+    "image_ref": <str | null>,
+    "text": <str>,
+    "user": {
+        "location": [<double>, <double>],
+        "name": <str | null>
+    }
+}
+```
+
+#### Response:
+
+```json
+[
+    {
+        "area": {
+            "coordinates": [
+                [
+                    [<double>, <double>],
+                    ...
+                ]
+            ],
+            "type": "Polygon"
+        },
+        "country": <str>,
+        "created": <long>,
+        "id": <str>,
+        "image_ref": <str | null>,
+        "like_count": <int>,
+        "text": <str>,
+        "user": {
+            "location": [<double>, <double>],
+            "name": <str | null>
+        }
+    },
+    ...
+]
+```
+
+Returns `201` if the post was successfully created.
+
+### `GET /posts`
+
+Retrieve a list of all posts.
+
+#### Response:
+
+```json
+[
+    {
+        "area": {
+            "coordinates": [
+                [
+                    [<double>, <double>],
+                    ...
+                ]
+            ],
+            "type": "Polygon"
+        },
+        "country": <str>,
+        "created": <long>,
+        "id": <str>,
+        "image_ref": <str | null>,
+        "like_count": <int>,
+        "text": <str>,
+        "user": {
+            "location": [<double>, <double>],
+            "name": <str | null>
+        }
+    },
+    ...
+]
+```
+
+### `PUT /post/<id>`
+
+Update an existing post with given `id`.
+
+#### Request:
+
+```json
+{
+    "text": <str>,
+}
+```
+
+#### Response:
+
+```json
+{
+    "area": {
+        "coordinates": [
+            [
+                [<double>, <double>],
+                ...
+            ]
+        ],
+        "type": "Polygon"
+    },
+    "country": <str>,
+    "created": <long>,
+    "id": <str>,
+    "image_ref": <str | null>,
+    "like_count": <int>,
+    "text": <str>,
+    "user": {
+        "location": [<double>, <double>],
+        "name": <str | null>
+    }
+}
+```
+
+Returns `200` if the post was successfully updated.
```

Comment:
..hm don't know if we should make it explizit the whole time. I think from here on it is ok.